pkg/broker/logic.go

```diff
@@ -39,6 +42,33 @@ func (b *BusinessLogic) GetCatalog(c *broker.RequestContext) (*broker.CatalogRes
 	return response, nil
 }
 
+func (b *BusinessLogic) ActionRotateCredentials(InstanceID string, vars map[string]string, context *broker.RequestContext) (interface{}, error) {
+	instance, err := b.GetInstanceById(InstanceID)
+	if err != nil {
+		return nil, NotFound()
+	}
+
+	provider, err := GetProviderByPlan(b.namePrefix, instance.Plan)
+	if err != nil {
+		glog.Errorf("Unable to rotate access keys, cannot find provider (GetProviderByPlan failed): %s\n", err.Error())
+		return nil, InternalServerError()
+	}
+
+	user, err := provider.RotateCredentials(instance)
+	if err != nil {
+		glog.Errorf("Unable to rotate access keys, RotateCredentials failed: %s\n", err.Error())
+		return nil, InternalServerError()
+	}
+
+	err = b.storage.UpdateCredentials(instance, user)
+	if err != nil {
```

Comment:
Are we going to somehow indicate that the older credentials are deprecated or remove the old creds?

Reply:
You're right in that it won't restart the app or any apps that the s3 is attached too.  

We could add an additional action to list out all the access key id's and indicate which one is the one being used, then add another option to destroy a specific access key id.  That way the end user on the CLI can first rotate it, take whatever steps necessary (like restart an app) then go back and remove the old key id's once everything is up and working correctly.

Thoughts?

Reply:
yeah, as long as we are ok with the manual step to remove.  If it won't restart the app, the app will break anyway if it invalidates, so i'm almost tempted to delete the key, but there would be no rollback. So the options are, delete the old key outright, or then add a manual step in. You can only have two keys existing at a time, so they'd have to delete the old key before rotating again.